components/suggest/src/db.rs

```diff
@@ -322,101 +361,163 @@ impl<'a> SuggestDao<'a> {
         let (keyword_prefix, keyword_suffix) = split_keyword(keyword_lowercased);
-        let suggestions = self.conn.query_rows_and_then_cached(
-            "SELECT s.id, MAX(k.rank) AS rank, s.title, s.url, s.provider, s.score, k.confidence, 
-                k.keyword_suffix
-            FROM suggestions s
-            JOIN prefix_keywords k ON k.suggestion_id = s.id
-            WHERE k.keyword_prefix = :keyword_prefix AND
-                k.keyword_suffix BETWEEN :keyword_suffix AND :keyword_suffix || x'FFFF' AND
-                s.provider = :provider
-            GROUP BY s.id, k.confidence
-            ORDER BY s.score DESC, rank DESC",
-             named_params! {
-                ":keyword_prefix": keyword_prefix,
-                ":keyword_suffix": keyword_suffix,
-                ":provider": SuggestionProvider::Pocket,
-            },
-            |row| -> Result<Option<Suggestion>>{
-                let title = row.get("title")?;
-                let raw_url = row.get::<_, String>("url")?;
-                let score = row.get::<_, f64>("score")?;
-                let confidence = row.get("confidence")?;
-                let full_suffix = row.get::<_, String>("keyword_suffix")?;
-                let suffixes_match = match confidence {
-                    KeywordConfidence::Low => full_suffix.starts_with(keyword_suffix),
-                    KeywordConfidence::High => full_suffix == keyword_suffix,
-                };
-                if suffixes_match {
-                    Ok(Some(Suggestion::Pocket {
-                        title,
-                        url: raw_url,
-                        score,
-                        is_top_pick: matches!(
-                        confidence,
-                        KeywordConfidence::High)
-                    }))
-                } else {
-                    Ok(None)
-                }
-            }
-            )?.into_iter().flatten().take(
-                query.limit.and_then(|limit|
-                    usize::try_from(limit).ok()
-                ).unwrap_or(usize::MAX)
-            ).collect();
+        let suggestions = self
+            .conn
+            .query_rows_and_then_cached(
+                r#"
+            SELECT
+              s.id,
+              MAX(k.rank) AS rank,
+              s.title,
+              s.url,
+              s.provider,
+              s.score,
+              k.confidence,
+              k.keyword_suffix
+            FROM
+              suggestions s
+            JOIN
+              prefix_keywords k
+              ON k.suggestion_id = s.id
+            WHERE
+              k.keyword_prefix = :keyword_prefix
+              AND k.keyword_suffix
+              BETWEEN :keyword_suffix
+              AND :keyword_suffix || x'FFFF'
+              AND s.provider = :provider
+            GROUP BY
+              s.id,
+              k.confidence
+            ORDER BY
+              s.score DESC,
+              rank DESC
+            "#,
```

Comment:
```suggestion
                r#"
                SELECT
                  s.id,
                  MAX(k.rank) AS rank,
                  s.title,
                  s.url,
                  s.provider,
                  s.score,
                  k.confidence,
                  k.keyword_suffix
                FROM
                  suggestions s
                JOIN
                  prefix_keywords k
                  ON k.suggestion_id = s.id
                WHERE
                  k.keyword_prefix = :keyword_prefix
                  AND (k.keyword_suffix BETWEEN :keyword_suffix AND :keyword_suffix || x'FFFF')
                  AND s.provider = :provider
                GROUP BY
                  s.id,
                  k.confidence
                ORDER BY
                  s.score DESC,
                  rank DESC
                "#,
```

Nit: Let's align the left edge with the `r#`, for consistency with the others—and parenthesize the `BETWEEN`.